v2/pkg/engine/plan/configuration_visitor.go

```diff
@@ -801,6 +804,101 @@ func (c *configurationVisitor) addNewPlanner(ref int, typeName, fieldName, curre
 	return len(c.planners) - 1, true
 }
 
+func (c *configurationVisitor) resolveSubscriptionFilterCondition(typeName, fieldName string) *resolve.SubscriptionFilter {
+	fieldConfig := c.fieldConfigurations.ForTypeField(typeName, fieldName)
+	if fieldConfig == nil {
+		return nil
+	}
+	if fieldConfig.SubscriptionFilterCondition == nil {
+		return nil
+	}
+	return c.buildSubscriptionFilterCondition(fieldConfig.SubscriptionFilterCondition)
+}
+
+func (c *configurationVisitor) buildSubscriptionFilterCondition(condition *SubscriptionFilterCondition) *resolve.SubscriptionFilter {
+	filter := &resolve.SubscriptionFilter{}
+	if condition.And != nil {
+		for _, andCondition := range condition.And {
+			filter.And = append(filter.And, c.buildSubscriptionFilterCondition(andCondition))
+		}
+	}
+	if condition.Or != nil {
+		for _, orCondition := range condition.Or {
+			filter.Or = append(filter.Or, c.buildSubscriptionFilterCondition(orCondition))
+		}
+	}
+	if condition.Not != nil {
+		filter.Not = c.buildSubscriptionFilterCondition(condition.Not)
+	}
+	if condition.In != nil {
+		filter.In = c.buildSubscriptionFieldFilter(condition.In)
+	}
+	if filter.And == nil && filter.Or == nil && filter.Not == nil && filter.In == nil {
+		return nil
+	}
+	return filter
+}
+
+var (
+	subscriptionFieldFilterRegex = regexp.MustCompile(`{{\s*args\.([a-zA-Z0-9_]+)\s*}}`)
+)
+
+func (c *configurationVisitor) buildSubscriptionFieldFilter(condition *SubscriptionFieldCondition) *resolve.SubscriptionFieldFilter {
+	filter := &resolve.SubscriptionFieldFilter{}
+	filter.FieldPath = condition.FieldPath
+	filter.Values = make([]resolve.InputTemplate, len(condition.Values))
+	for i, value := range condition.Values {
+		matches := subscriptionFieldFilterRegex.FindAllStringSubmatchIndex(value, -1)
+		if len(matches) == 0 {
+			filter.Values[i].Segments = []resolve.TemplateSegment{
+				{
+					SegmentType: resolve.StaticSegmentType,
+					Data:        []byte(value),
+				},
+			}
+			continue
+		}
+		if len(matches) == 1 && len(matches[0]) == 4 {
+			prefix := value[:matches[0][0]]
```

Comment:
could you add some explanations about indexes? [0][2] not so clear why it is like this

Reply:
It's what the std Library gives back.
One match with 2 sub matches is a slice of one with a slice of 4, two indices marking start and end of the match.

Reply:
please add input examples which regexp should match